includes/class-convertkit-ajax.php

```diff
@@ -99,6 +99,16 @@ public function store_subscriber_email_as_id_in_cookie() {
 		}
 		$email = sanitize_text_field( $_REQUEST['email'] );
 
+		// Bail if the email address is empty.
+		if ( empty( $email ) ) {
+			wp_send_json_error( __( 'ConvertKit: Required parameter `email` is empty.', 'convertkit' ) );
+		}
+
+		// Bail if the email address isn't a valid email address.
+		if ( ! filter_var( $email, FILTER_VALIDATE_EMAIL ) ) {
+			wp_send_json_error( __( 'ConvertKit: Required parameter `email` is not an email address.', 'convertkit' ) );
+		}
```

Comment:
Email addresses are notoriously difficult to validate and I'm wondering if a simple check for `@` would be enough to avoid unnecessary requests. I wouldn't want to lose a subscriber if the validation filter is wrong.

Reply:
So you don't "lose" a subscriber here; the ConvertKit hosted Form will always submit, and ConvertKit will always run its own validation, and subscribe the email address as necessary.

The logic here is to prevent the separate Plugin-specific AJAX request from hitting the `subscribers` endpoint to fetch a subscriber ID by email address.  It's not critical to get a subscriber ID, more of a convenience so it's stored as a cookie for possible later use for e.g. showing content based on the existence of a subscriber's tag.